### ISSUE TITLE
[bug] [4.x] Hidden block types can still be added via the settings "cog" icon on existing blocks

### ISSUE THREAD

**Issue by tomdavies** [angell-co/Spoon#122]
Where Spoon has been configured to hide some Matrix blocks for a field layout, these hidden block types can still be added to the field still be via the settings "cog" icon on existing blocks:

<img width="865" alt="Screenshot 2022-10-11 at 12 21 53" src="https://user-images.githubusercontent.com/136308/195078022-897521df-966e-4cac-8cfb-19aebf210311.png">

The items in that list of matrix block types have a `data-type="<blockHandle>"` attribute on them, so it should be fairly easy to find that list and hide the block types that are not enabled for this field layout

<img width="1435" alt="Screenshot 2022-10-11 at 12 27 26" src="https://user-images.githubusercontent.com/136308/195078148-775f41c0-92e2-42b8-8b89-ab6fa7d99502.png">

LMK if you would accept a PR for this!

**Comment by tigerchick** [comment]
plus 1  for this. 
This recently caused me a little trouble with a client adding blocks they weren't meant to be able to!

**Comment by tomdavies** [comment]
Having dug into this some more, there is code that should prevent this, but it never fires.

At the start of `FieldManipulator::initSettingsMenu()` we have:

```js
Garnish.requestAnimationFrame($.proxy(function()
    {
    // Get the Garnish.MenuBtn object
    var menuBtn = $settingsBtn.data('menubtn') || false;
    
    // If there wasn’t one then fail and try again
    if (!menuBtn)
    {
        this.initSettingsMenu($settingsBtn, spoonedBlockTypes, $matrixField);
        return;
    }
...
```

The problem is `$settingsBtn.data('menubtn')` is _always_ `false`. Guessing this is due to a change to the edit screen JS in Craft 4

**Comment by tomdavies** [comment]
Alright, I figured this out. The settings menu now uses `Garnish.DisclosureMenu` rather than the `MenuBtn` it used before. Fix is just a small JS change, PR inbound...

**Comment by tomdavies** [comment]
PRed 😄 

 CC @tigerchick in case you're in a hurry for a fix

**Comment by tigerchick** [comment]
> PRed 😄
> 
> CC @tigerchick in case you're in a hurry for a fix

Thanks @tomdavies !

**Comment by lindseydiloreto** [comment]
PR has been merged, and the fix has been released in **v4.0.2**.

Thanks for all of your recent contributions @tomdavies! 🙏
